src/Polysemy/Internal/Combinators.hs

```diff
@@ -352,7 +351,8 @@ rewrite
 rewrite f (Sem m) = Sem $ \k -> m $ \u ->
```

Comment:
Please also change the type signatures for `rewrite` and `transform`. In contrast to `interpret`/`intercept`, this is actually important to do.